Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp

```diff
@@ -364,10 +484,55 @@ namespace AzToolsFramework
         DPEModelNode* theNode = (parent.isValid() ? GetNodeFromIndex(parent) : m_rootNode);
         if (theNode)
         {
-            rowCount = theNode->GetChildCount();
+            rowCount = theNode->GetRowChildCount();
         }
         return rowCount;
     }
-} // namespace AzToolsFramework
 
-// #include "UI/moc_DPEDebugModel.cpp"
+    void DPEDebugModel::HandleReset()
+    {
+        beginResetModel();
+        if (m_rootNode)
+        {
+            m_rootNode->deleteLater(); // recursively destroys the existing tree
+        }
+
+        // recursively populate the node tree with the new adapter information
+        m_rootNode = new DPEModelNode(this);
+        m_rootNode->SetValue(m_adapter->GetContents(), false);
```

Comment:
note to self: try just setting the existing root node, I don't think we need to destroy it to set it.

Reply:
this worked, and is simpler. Fixed.